yaksh/views.py

```diff
@@ -494,20 +499,28 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None):
             # validation
             if 'assignment' in request.FILES:
                 assignment_filename = request.FILES.getlist('assignment')
+                qp = QuestionPaper.objects.get(id=questionpaper_id)
             for fname in assignment_filename:
                 if AssignmentUpload.objects.filter(
                     assignmentQuestion=current_question,
-                    assignmentFile__icontains=fname, user=user).exists():
+                    assignmentFile__icontains=fname, user=user,
```

Comment:
If we save this query in a variable like assignment_files, then check assignment_files.exists()
Also, on assignment_files you can further query to get your "assign_file".
This will reduce lot of code and slight redundancy.